trtools/utils/tr_harmonizer.py

```diff
@@ -1167,12 +1182,18 @@ def GetDosages(self,
             else:
                 # Normalize to be between 0 and 2
                 dosages = (unnorm_dosages-2*self.min_allele_length)/(self.max_allele_length-self.min_allele_length)
-                assert not (np.any(dosages>=2.1) or np.any(dosages<=-0.1))
+                if (np.any(dosages>=2.1) or np.any(dosages<=-0.1)):
+                    error_msg = "{}:{} Error normalizing dosages: value >=2.1 or <=-0.1 detected".format(self.chrom, self.pos)
+                    if strict:
+                        raise ValueError(error_msg)
+                    else:
+                        common.WARNING(error_msg)
+                        return np.array([np.nan]*self.GetNumSamples())
```

Comment:
Same comment as above. If you couldn't compute dosages for some samples, could you still output dosages for the rest of them? Or would it be better to indicate complete uncertainty at this locus?
```suggestion
                failing = (dosages>=2.1) | (dosages<=-0.1)
                if np.any(failing):
                    error_msg = "{}:{} Error normalizing dosages: value >=2.1 or <=-0.1 detected".format(self.chrom, self.pos)
                    if strict:
                        raise ValueError(error_msg)
                    else:
                        common.WARNING(error_msg)
                        dosages[failing] = np.nan
```